### PR TITLE
Implement initial notification provider

### PR DESCRIPTION
**Pull request by BrainBalaka**
## Summary
Initial implementation of notification provider. One notification provider is implemented: Email. Further capability of email will be added later, such as using razorview to generate the email body.

### CODE REVIEW

**Comment by BrainBalaka** [comment]
@frandi I've updated the PR to:
- make the NotificationProvider have 1 generic method, and passing the message type in the request
- make the message subject configurable
- make the message body retrieved from template folder, and have a generic and sender specific template available.

Can you please take a look? Thanks

File: src/Shared/Polyrific.Catapult.Shared.Common/Interface/INotificationSender.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Polyrific, Inc 2018. All rights reserved.
+
+using Polyrific.Catapult.Shared.Common.Notification;
+
+namespace Polyrific.Catapult.Shared.Common.Interface
+{
+    public interface INotificationSender
```
Comment (frandi) [inline src/Shared/Polyrific.Catapult.Shared.Common/Interface/INotificationSender.cs]: Let's complete the props and methods with description comments.

File: src/Shared/Polyrific.Catapult.Shared.SmtpEmailNotification/SmtpSetting.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Polyrific, Inc 2018. All rights reserved.
+
+namespace Polyrific.Catapult.Shared.SmtpEmailNotification
+{
+    public class SmtpSetting
```
Comment (frandi) [inline src/Shared/Polyrific.Catapult.Shared.SmtpEmailNotification/SmtpSetting.cs]: Let's complete this with desc comments as well.

File: src/Shared/Polyrific.Catapult.Shared.SmtpEmailNotification/Polyrific.Catapult.Shared.SmtpEmailNotification.csproj [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```
Comment (frandi) [inline src/Shared/Polyrific.Catapult.Shared.SmtpEmailNotification/Polyrific.Catapult.Shared.SmtpEmailNotification.csproj]: Can you try change this to `netstandard2.0` moniker and see if it works?

Comment (BrainBalaka) [reply src/Shared/Polyrific.Catapult.Shared.SmtpEmailNotification/Polyrific.Catapult.Shared.SmtpEmailNotification.csproj]: yes, it works. I'll change to `netstandard2.0`

**Pull request merged**